modules/developing/pages/clients/flink.adoc

```diff
@@ -0,0 +1,476 @@
+A Flink *job* is a collection of one or more Flink applications that are submitted to a Flink cluster for execution. In a Flink job, data is typically ingested from various sources such as Kafka, HDFS, or databases. The data is then processed using Flink's high-level APIs or low-level operators, which provide a wide range of transformations and operations for data manipulation, filtering, aggregation, and analytics.
+
+Overall, a Flink job enables you to process and analyze large-scale data sets efficiently, with low-latency and high throughput, making it suitable for various use cases such as real-time analytics, event-driven applications, data pipelines, and ETL (Extract, Transform, Load) processes.
+
+A little more setup is required to create a Flink job.
+
+It is VERY important that on the Pulsar side (in {product_name}, in this case) you have pre-created both the input and output topics, and that all the topics have a schema that is compatible with the Flink table schema (in this case, a JSON schema).
+
+. First, add more environment variables to complete this example:
++
+[source,bash]
+----
+PULSAR_CLIENT_CONF=$PULSAR_HOME/conf/clienf.conf
+inputTopic=flink-tenant/flink/persons
+outputTopic=flink-tenant/flink/persons-output
+PULSAR_HOME=$(pwd)
+ADMINCMD=$PULSAR_HOME/bin/pulsar-admin
+CLIENTCMD=$PULSAR_HOME/bin/pulsar-client
+$ADMINCMD topics delete-partitioned-topic $inputTopic
+$ADMINCMD topics delete-partitioned-topic $outputTopic
+$ADMINCMD topics create-partitioned-topic -p 2 $inputTopic
+$ADMINCMD topics create-partitioned-topic -p 2 $outputTopic
+----
+
+. Produce data to set a JSON schema on the "persons" (`$inputTopic`) and "persons-output" (`$outputTopic`) topics.
+First, we'll produce messages to the "persons" (`$inputTopic`) topic:
++
+[tabs]
+====
+Curl::
++
+--
+[source,bash]
+----
+$CLIENTCMD produce -k family-1 -m "{\"name\":\"giovanni\"}","{\"name\":\"enrico\"}","{\"name\":\"giovanni\"}" -vs 'json:{"type": "record","namespace":"com.example","name":"Person","fields":[{"name":"name","type":"string"}]}'  $inputTopic
+----
+--
+
+Result::
++
+--
+[source,bash]
+----
+...
+2023-05-24T17:35:21,391-0400 [main] INFO  org.apache.pulsar.client.cli.PulsarClientTool - 3 messages successfully produced
+----
+--
+====
++
+.. We produced three messages with JSON schema to the "persons" (`$inputTopic) topic. Let's confirm our schema looks as expected:
++
+[tabs]
+====
+Curl::
++
+--
+[source,bash]
+----
+$ADMINCMD schemas get $inputTopic
+----
+--
+
+Result::
++
+--
+[source,json]
+----
+{
+  "version": 0,
+  "schemaInfo": {
+    "name": "persons",
+    "schema": {
+      "type": "record",
+      "namespace": "com.example",
+      "name": "Person",
+      "fields": [
+        {
+          "name": "name",
+          "type": "string"
+        }
+      ]
+    },
+    "type": "JSON",
+    "timestamp": 1684964118894,
+    "properties": {}
+  }
+}
+----
+--
+====
+
+.. Now, we'll produce messages to the "persons-output" (`$outputTopic`) topic:
```

Comment:
Not sure why we need to produce a message to this topic at this point? Is it supposed to be written by the Flink job?